gradio-app/gradio
# Fix bugs with gr.update

# Description

Fixes #2154 #2156

During #2044, I made it so that `delete_none` would keep `None` in the `value` key. This was to let users set the value of the component to `null` in the front-end which would reset the component value. The problem is that `None` is the default of the `value` key in the `update` component method, so the value is always reset, even when users are not trying to reset the value!!

The solution I went for was to distinguish the "don't update the value" and "set the value to null" use cases. For the former, we use an enum for the default value of the update method and for the latter, the user can return `None` as before.

The enum I chose is `_Keywords.NO_VALUE`. It's not exposed at the top-level gradio package and it's marked "private" by the leading underscore so it should not be autocompleted in IDES. So it should mean that the user did not pass in a value for the `"value"` key.

To test:

```python
import gradio as gr

with gr.Blocks() as demo:
    radio = gr.Radio(label="algorithm", choices=["a", "b", "c"], value="a")
    slider = gr.Slider(label="threshold", value=0, minimum=-1, maximum=1, visible=False, interactive=True)
    style = gr.Dropdown(label="style", choices=["modern", "classical", "cartoon"], value="modern", visible=False, interactive=True)

    def make_visible(algorithm):
        is_visible = algorithm == "c"
        return {slider: gr.update(visible=is_visible), style: gr.update(visible=is_visible)}
    radio.change(make_visible, inputs=radio, outputs=[slider, style])

demo.launch()
```
The slider value and drop down should be persisted when clicking different radio buttons.

I also added `blocks_update` to `all_demos` - slider value should be persisted as well.

# Checklist:
